### PR TITLE
Add map type property (standard, satellite, hybrid) to MapView.

### PR DESCRIPTION
**Pull request by adamkrell**
Here is an example project demonstrating this pull request: [AKMapView](https://github.com/adamkrell/AKMapView)

### CODE REVIEW

**Comment by paramaggarwal** [comment]
I suggest using `enum`s instead of `NSUInteger`. Take a look at how the `Image` component [RCTNetworkImageViewManager.m](https://github.com/facebook/react-native/blob/master/Libraries/Image/RCTNetworkImageViewManager.m) uses the `UIViewContentMode` enum with help from [RCTConvert.h](https://github.com/facebook/react-native/blob/master/React/Base/RCTConvert.h).

**Comment by adamkrell** [comment]
@paramaggarwal Thanks for pointing me in the right direction. I updated the PR to use an enum for the mapType property. I also updated my example project: [AKMapView](https://github.com/adamkrell/AKMapView)

File: React/Views/RCTMapManager.m [hunk 1]
```diff
@@ -41,6 +41,7 @@ - (UIView *)view
 RCT_EXPORT_VIEW_PROPERTY(maxDelta, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(minDelta, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(legalLabelInsets, UIEdgeInsets)
+RCT_REMAP_VIEW_PROPERTY(mapType, mapType, MKMapType)
```
Comment (paramaggarwal) [inline React/Views/RCTMapManager.m]: You can leave this at `RCT_EXPORT_VIEW_PROPERTY(mapType, MKMapType)` as the implementation of `RCT_EXPORT_VIEW_PROPERTY` is anyways like this:

```objective-c
#define RCT_EXPORT_VIEW_PROPERTY(name, type) RCT_REMAP_VIEW_PROPERTY(name, name, type)
```

Comment (adamkrell) [inline React/Views/RCTMapManager.m]: @paramaggarwal I didn't realize that. Updated to use ` RCT_EXPORT_VIEW_PROPERTY`.

File: React/Views/RCTMap.m [hunk 2]
```diff
@@ -120,4 +120,8 @@ - (void)setAnnotations:(MKShapeArray *)annotations
   }
 }
 
+- (void)setMapType:(MKMapType)mapType
+{
+    [super setMapType:mapType];
```
Comment (brentvatne) [inline React/Views/RCTMap.m]: nit: this should be indented two spaces

**Comment by brentvatne** [comment]
@adamkrell - looks good aside from the one small style nit, can you fix that and squash into one commit?

**Comment by adamkrell** [comment]
@brentvatne Fixed coding convention issue and squashed commits.

**Comment by brentvatne** [comment]
Looks good to me!

**Comment by brentvatne** [comment]
@tadeuzagallo - assigned you to this as your name appears most frequently in related code, re-assign as you see fit of course :smile:

File: React/Views/RCTMap.m [hunk 3]
```diff
@@ -120,4 +120,8 @@ - (void)setAnnotations:(MKShapeArray *)annotations
   }
 }
 
+- (void)setMapType:(MKMapType)mapType
+{
+  [super setMapType:mapType];
```
Comment (tadeuzagallo) [inline React/Views/RCTMap.m]: If it just calls `super`, it's not needed.

**Comment by adamkrell** [comment]
@tadeuzagallo I removed the unnecessary changes to RCTMap.

**Comment by tadeuzagallo** [comment]
Looks good now, thanks! :smiley:

@facebook-github-bot import

**Pull request closed without merging**